x-pack/filebeat/input/azureblobstorage/input_stateless.go

```diff
@@ -65,10 +66,10 @@ func (in *statelessInput) Run(inputCtx v2.Context, publisher stateless.Publisher
 		st := newState()
 		currentSource := source.(*Source)
 		log := inputCtx.Logger.With("account_name", currentSource.AccountName).With("container", currentSource.ContainerName)
-		// create a new inputMetrics instance
-		metrics := newInputMetrics(inputCtx.ID+":"+currentSource.ContainerName, nil)
+		// use a new metrics registry associated to no parent. No metrics will
+		// be published.
+		metrics := newInputMetrics(monitoring.NewRegistry())
```

Comment:
Why is a new registry passed in here, when this is replacing a call that gave a nonempty ID? Shouldn't this one use the context's registry (and possibly override its ID field for backwards compatibility)?